examples/python/controlnet/README.md

```diff
@@ -1,15 +1,79 @@
 <!--[metadata]
 title = "ControlNet"
 tags = ["controlnet", "canny", "huggingface", "stable-diffusion", "tensor", "text"]
+description = "Use Hugging Face's ControlNet to condition Stable Diffusion on edges detected by the Canny edge detector."
 thumbnail = "https://static.rerun.io/controlnet/8aace9c59a423c2eeabe4b7f9abb5187559c52e8/480w.png"
 thumbnail_dimensions = [480, 303]
 -->
 
+<picture>
+  <source media="(max-width: 480px)" srcset="https://static.rerun.io/controlnet/8aace9c59a423c2eeabe4b7f9abb5187559c52e8/480w.png">
+  <source media="(max-width: 768px)" srcset="https://static.rerun.io/controlnet/8aace9c59a423c2eeabe4b7f9abb5187559c52e8/768w.png">
+  <source media="(max-width: 1024px)" srcset="https://static.rerun.io/controlnet/8aace9c59a423c2eeabe4b7f9abb5187559c52e8/1024w.png">
+  <source media="(max-width: 1200px)" srcset="https://static.rerun.io/controlnet/8aace9c59a423c2eeabe4b7f9abb5187559c52e8/1200w.png">
+  <img src="https://static.rerun.io/controlnet/8aace9c59a423c2eeabe4b7f9abb5187559c52e8/full.png" alt="">
+</picture>
 
-This example integrates Rerun into [Hugging Face's ControlNet example](https://huggingface.co/docs/diffusers/using-diffusers/controlnet#controlnet). ControlNet allows to condition Stable Diffusion on various modalities. In this example we condition on edges detected by the Canny edge detector.
+Use [Hugging Face's ControlNet](https://huggingface.co/docs/diffusers/using-diffusers/controlnet#controlnet) to condition Stable Diffusion on edges detected by the Canny edge detector.
+
+
+
+## Used Rerun Types
+[`Image`](https://www.rerun.io/docs/reference/types/archetypes/image), [`Tensor`](https://www.rerun.io/docs/reference/types/archetypes/tensor), [`TextDocument`](https://www.rerun.io/docs/reference/types/archetypes/text_document)
+
+
+## Background
+[Hugging Face's ControlNet](https://huggingface.co/docs/diffusers/using-diffusers/controlnet#controlnet). allows to condition Stable Diffusion on various modalities. In this example we condition on edges detected by the Canny edge detector to keep our shape intact while generating an image with Stable Diffusion.
 
 https://vimeo.com/870289439?autoplay=1&loop=1&autopause=0&background=1&muted=1&ratio=1440:1080
 
+# Logging and Visualizing with Rerun
+The visualizations in this example were created with the following Rerun code.
+
+## Canny image
+Uses cv2 to calculate a [canny image](https://docs.opencv.org/4.x/da/d22/tutorial_py_canny.html) that shows the edges in the image in black and white.
+```python
+canny_image = cv2.Canny(rgb_image, low_threshold, high_threshold)
+canny_image = canny_image[:, :, None]
+canny_image = np.concatenate([canny_image, canny_image, canny_image], axis=2)
+canny_image = PIL.Image.fromarray(canny_image)
```

Comment:
Regarding here and below in this section. The original idea was to just show the Rerun code needed to create the visualization and to send the reader to the code for more details. In this case it might actually make sense to have the other code as well since it's so simple but I don't know. What do you think? I want to make sure this page is still mainly about how to use Rerun to visualize but it is important that it's easy to understand.

Reply:
The fact that Rerun requires code for visualisation to work is important for users to understand. 

For instance for someone without any experience they might think that Rerun is a GUI that visualises your data without any code.

Still its more important for users to understand Rerun then other code in the examples.

Maybe different tutorials can highlight different aspects of Rerun and this can be a way to highlight information that is more important for users to understand, for instance in this tutorial, the timeless concept could be made more prominent.

I made an example of how that can work.